crates/modelardb_server/src/context.rs

```diff
@@ -356,4 +391,304 @@ impl Context {
         }
         Ok(())
     }
+
+    /// Return the schema of `table_name` if the table exists in the default database schema,
+    /// otherwise a [`ModelarDbError`] indicating at what level the lookup failed is returned.
+    pub async fn schema_of_table_in_default_database_schema(
+        &self,
+        table_name: &str,
+    ) -> Result<SchemaRef, ModelarDbError> {
+        let database_schema = self.default_database_schema()?;
+
+        let table = database_schema.table(table_name).await.ok_or_else(|| {
+            ModelarDbError::DataRetrievalError("Table does not exist.".to_owned())
```

Comment:
Include the table name in the error message?

Reply:
Good idea, fixed.